lib/workspace.rb

```diff
@@ -0,0 +1,76 @@
+require_relative "user"
+require_relative "channel"
+require "pry"
+
+module SlackCLI
+  class SlackApiError < Exception; end
+
+  class Workspace
+    attr_reader :users, :channels, :selected, :selected, :query
+
+    Dotenv.load
+
+    def initialize()
+      @query = { token: SLACK_TOKEN }
+      @users = self.list_all_user
+      @channels = self.list_all_channel
+      @selected = nil
+      @selected = nil
+    end
+
+    def list_all_user
+      all_users = []
+      response = HTTParty.get(SlackCLI::User::BASE_URL, query: query)["members"]
+
+      response.each do |person|
+        all_users << SlackCLI::User.new(person["name"], person["id"])
+      end
+      return all_users
+    end
```

Comment:
Methods like this for user and Channel could be instead class methods in the User and Channel class and that would help separate the command-line stuff in workspace with the HTTP requests in Recipient, Channel and User.